### PR TITLE
Fix overflowing user email address

### PR DESCRIPTION
**Pull request by vickywane**
Resolves [this](https://github.com/opencollective/opencollective/issues/3662) issue

# Description
 This PR contains an added props in the `TopBarProfileMenu` component which prevents the logged-in user's email address from overflowing. Using `word-break : break-all` , it breaks the overflowing part of the email into a new line.

### CODE REVIEW

File: components/Text.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ export const P = styled.p.attrs(props => ({
       white-space: nowrap;
       overflow: hidden;
     `}
+  ${props => props.wordBreak}
```
Comment (Betree) [inline components/Text.js:26]: `wordBreak` prop is already supported by this component, see https://github.com/opencollective/opencollective-frontend/pull/5395/files#diff-483920b66c212791608965bc62aa404cd8b73051c25192b68a856cd6ecb4e7b4R17

**Review by Betree — CHANGES_REQUESTED** [review]
There are some issues with prettier, could you check?

File: components/Text.js (hunk 1 above)
Comment (vickywane) [reply components/Text.js:26]: Alright. I've taken note of it.

**Comment by vickywane** [comment]
@Betree running `npm run lint:fix` logs out `17 errors` and `68 warnings` and I don't think they are from my fix :).

**Review by SudharakaP — COMMENTED** [review]
@vickywane : What you have problems with is prettier. Prettier is used in these projects for formatting. Now `npm run lint:fix` will not fix prettier issues. To fix those you need to run `npm run prettier:write`. 😄

**Comment by vickywane** [comment]
Aha thanks. 

I have done that.

**Comment by SudharakaP** [comment]
> Aha thanks.
> 
> I have done that.
> 
> @SudharakaP would it be okay to rebase the commits since they have already been pushed?

@vickywane : I don't see you've done that; maybe you've not pushed it yet? The CI for prettier is still failing. 🤔 

Yes what I normally do is an [interactive re-base](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase) at the end followed by a force push so that you will have one single commit.

**Comment by vickywane** [comment]
I haven't pushed yet.

**Comment by vickywane** [comment]
@SudharakaP I mistakenly rebased off a commit higher than my first commit then I added a new commit to make the circle-ci run the tests but it seems that won't work. 

Can the tests be manually rerun or should I close this PR and use the GitHub editor to make the change since it's a small one-line fix?

**Comment by SudharakaP** [comment]
@vickywane: I would say just don't worry about it as I can only see changes
relevant to you in the PR and the person who merge can always squash and
merge. In the future maybe you could try out interactive rebasing; although
I don't think it's something strictly adhered to in this repo (maybe
@Betree can verify).

**Comment by vickywane** [comment]
Alright. I used the interactive rebasing but I counted the commits wrongly.

**Comment by Betree** [comment]
> In the future maybe you could try out interactive rebasing; although I don't think it's something strictly adhered to in this repo

@SudharakaP Indeed, we don't require commits to be squashed but we usually squash & merge if they aren't.

**Review by Betree — APPROVED** [review]
It's working fine. Thanks @vickywane!

**Comment by vickywane** [comment]
Yay. 

Thank you

**Pull request merged**